### PR TITLE
allow input plugins to be py3 compat

### LINKED ISSUE #26278: Pause prompt broken in python3
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
pause

##### ANSIBLE VERSION

```
ansible 2.3.2.0 (detached HEAD 7424e1c417) last updated 2017/06/30 11:12:33 (GMT +200)
  config file = 
  configured module search path = Default w/o overrides
  python version = 3.6.1 (default, Apr  4 2017, 09:40:21) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.38)]

```

##### CONFIGURATION
Default

##### OS / ENVIRONMENT
MacOS

##### SUMMARY
Using the `prompt` functionality of the `pause` module, the interactive input does not stop anymore when pressing Enter. If the input is interrupted using Ctrl+C and the process is continued by pressing "C", the returned `user_input` also contains the newline characters.

##### STEPS TO REPRODUCE

```yaml
---

- name: Input
  pause:
    prompt: "Username?"
  register: r_prompt

- debug:
    msg: "{{r_prompt.user_input}}"
```

- Run this tasks and enter `123<ENTER>456`
- Because process does not continue, press CTRL+C and then C

##### EXPECTED RESULTS
 - expect process to continue when pressing <ENTER>
 - expect `user_input` to be `123`

##### ACTUAL RESULTS

```
TASK [basis : Input] ************************************************************************************************************************************************************************************************************************************************
task path: /Users/phsc/Projects/sandbox/test/roles/basis/tasks/test.yml:3
[basis : Input]
Username?:
Press 'C' to continue the play or 'A' to abort 
ok: [localhost] => {
    "changed": false,
    "delta": 5,
    "rc": 0,
    "start": "2017-06-30 11:21:19.867562",
    "stderr": "",
    "stdout": "Paused for 0.1 minutes",
    "stop": "2017-06-30 11:21:25.623051",
    "user_input": "123\n456"
}

TASK [basis : debug] ************************************************************************************************************************************************************************************************************************************************
task path: /Users/phsc/Projects/sandbox/test/roles/basis/tasks/test.yml:8
ok: [localhost] => {
    "msg": "123\n456"
}
META: ran handlers
META: ran handlers

```

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

fixes #26278

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

python3

##### ANSIBLE VERSION

```
2.4
```

### CODE REVIEW

File: lib/ansible/executor/process/worker.py [hunk 1]
```diff
@@ -77,7 +79,10 @@ def __init__(self, rslt_q, task_vars, host, task, play_context, loader, variable
                 fileno = sys.stdin.fileno()
                 if fileno is not None:
                     try:
-                        self._new_stdin = os.fdopen(os.dup(fileno))
+                        if PY3: # needed for recognizing Enter in 'pause' module
+                            self._new_stdin = os.fdopen(os.dup(fileno), newline='\r\n')
```
Comment (abadger) [inline lib/ansible/executor/process/worker.py:83]: Is the intention to *only* recognize Enter if it sends the string "\r\n"?  Or is the intention to recognize any of "\r", "\n", or "\r\n"?  Documentation says ( https://docs.python.org/3/library/functions.html#open ) that newline=None or newline="" should be used if the latter is meant.

Comment (bcoca) [reply lib/ansible/executor/process/worker.py:83]: @abadger I followed the original ticket and it seems to work as per tests, not sure of what other side effects we should look for

Comment (philsch) [reply lib/ansible/executor/process/worker.py:83]: To fix #26278 the intention is to deactivate "universal newlines mode" to exit the prompt correctly.

- when **newline=None** (default): Prompt is never exiting, ctrl+c is needed to interrupt
- when **newline=""**: prompt only exiting when pressing Enter and again any key at the keyboard
- when **newline is \n, \r or \r\n**: prompt is exiting as expected

First case is clear imho, all newline like symbols are translated to \n and stdin will never exit.
Second case is more tricky as I don't have that deep knowledge about python core functions. Maybe "universal newlines mode" is buffering single newline chars somehow and messing up the stdin?

**Comment by bcoca** [comment]
closing in favor of #26495

**Pull request closed without merging**